new-lamassu-admin/src/pages/Services/schemas/bitgo.js

```diff
@@ -80,47 +82,49 @@ export default {
       component: SecretInput
     }
   ],
-  validationSchema: Yup.object().shape({
-    token: Yup.string()
-      .max(100, 'Too long')
-      .required(),
-    BTCWalletId: Yup.string().max(100, 'Too long'),
-    BTCWalletPassphrase: Yup.string()
-      .max(100, 'Too long')
-      .when('BTCWalletId', {
-        is: isDefined,
-        then: Yup.string().required()
-      }),
-    LTCWalletId: Yup.string().max(100, 'Too long'),
-    LTCWalletPassphrase: Yup.string()
-      .max(100, 'Too long')
-      .when('LTCWalletId', {
-        is: isDefined,
-        then: Yup.string().required()
-      }),
-    ZECWalletId: Yup.string().max(100, 'Too long'),
-    ZECWalletPassphrase: Yup.string()
-      .max(100, 'Too long')
-      .when('ZECWalletId', {
-        is: isDefined,
-        then: Yup.string().required()
-      }),
-    BCHWalletId: Yup.string().max(100, 'Too long'),
-    BCHWalletPassphrase: Yup.string()
-      .max(100, 'Too long')
-      .when('BCHWalletId', {
-        is: isDefined,
-        then: Yup.string().required()
-      }),
-    DASHWalletId: Yup.string().max(100, 'Too long'),
-    DASHWalletPassphrase: Yup.string()
-      .max(100, 'Too long')
-      .when('DASHWalletId', {
-        is: isDefined,
-        then: Yup.string().required()
-      }),
-    environment: Yup.string()
-      .matches(/(prod|test)/)
-      .required()
-  })
+  getValidationSchema: account => {
+    return Yup.object().shape({
+      token: Yup.string()
+        .max(100, 'Too long')
+        .required(),
+      BTCWalletId: Yup.string().max(100, 'Too long'),
+      BTCWalletPassphrase: Yup.string()
+        .max(100, 'Too long')
+        .when('BTCWalletId', {
+          is: isDefined,
+          then: Yup.string().test(secretTest(account?.BTCWalletPassphrase))
+        }),
```

Comment:
Use a pure function to generate these records to reduce boilerplate. Maybe we can even replace the whole ``.when`` part.